homeassistant/components/tplink/camera.py

```diff
@@ -0,0 +1,221 @@
+"""Support for TPLink camera entities."""
+
+import asyncio
+from dataclasses import dataclass
+import datetime
+import logging
+
+from aiohttp import web
+from haffmpeg.camera import CameraMjpeg
+from kasa import Credentials, Device, Module
+from kasa.smartcam.modules import Camera as CameraModule
+
+from homeassistant.components import ffmpeg, stream
+from homeassistant.components.camera import (
+    Camera,
+    CameraEntityDescription,
+    CameraEntityFeature,
+)
+from homeassistant.config_entries import ConfigFlowContext
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.aiohttp_client import async_aiohttp_proxy_stream
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from . import TPLinkConfigEntry, legacy_device_id
+from .const import CONF_CAMERA_CREDENTIALS
+from .coordinator import TPLinkDataUpdateCoordinator
+from .entity import CoordinatedTPLinkEntity, TPLinkModuleEntityDescription
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkCameraEntityDescription(
+    CameraEntityDescription, TPLinkModuleEntityDescription
+):
+    """Base class for camera entity description."""
```

Comment:
Do we need to subclass here since we aren't adding new items, or is that planned in the future?